app/src/main/java/com/connorlay/knitgrid/adapters/KnitGridAdapter.java

```diff
@@ -0,0 +1,90 @@
+package com.connorlay.knitgrid.adapters;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.connorlay.knitgrid.R;
+import com.connorlay.knitgrid.models.Stitch;
+import com.squareup.picasso.Picasso;
+
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+
+/**
+ * Created by Greg on 3/17/2016.
+ */
+public class KnitGridAdapter extends RecyclerView.Adapter<KnitGridAdapter.StitchRecycleViewHolder> {
+
+
+    private final OnStitchRowClickListener mListener;
+
+    private List<Stitch> mStitches;
+
+    public interface OnStitchRowClickListener {
+        void onStitchRowClick(Stitch stitch);
+    }
+
+    public KnitGridAdapter(List<Stitch> stitches, OnStitchRowClickListener listener) {
+        mListener = listener;
+        mStitches = stitches;
+    }
+
+    @Override
+    public StitchRecycleViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View row = inflater.inflate(R.layout.legend_row, parent, false);
+        return new StitchRecycleViewHolder(row);
+    }
+
+    @Override
+    public void onBindViewHolder(StitchRecycleViewHolder holder, int position) {
+
+        final Stitch stitch = mStitches.get(position);
+        holder.textView.setText(stitch.getAbbreviation());
+
+
+        Context context = holder.imageView.getContext();
+
+        Picasso.with(context).load(stitch.getIconID()).fit().centerInside().into(holder.imageView);
+        holder.fullView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mListener != null) {
```

Comment:
I would consider using a [guard clause](http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) here to avoid an unnecessary indentation level.